packages/core-utils/src/query.js

```diff
@@ -209,9 +209,11 @@ export function planParamsToQuery(params, config) {
       case "time":
         query.time = params.time || getCurrentTime(config);
         break;
-      default:
-        if (!Number.isNaN(params[key])) query[key] = parseFloat(params[key]);
-        else query[key] = params[key];
+      default: {
+        const maybeNumber = parseFloat(params[key]);
+        query[key] = Number.isNaN(maybeNumber) ? params[key] : maybeNumber;
+        break;
+      }
```

Comment:
I'm worried about how this will handle strings that begin with a number character. E.g., let's say we're handling bannedRoutes, which is a comma-separated list of route IDs consisting of one route `897ABC`:
```
parseFloat('897ABC') // yields the number 897
Number.isNaN(897) // false
query[key] = 897 // instead of the correct string value 897ABC 
```
I think the original issue stated that OTP-RR used the global `isNaN` here, and I would perhaps advocate going back to that:
```
if (!isNaN('897ABC')) query[key] = parseFloat(params[key]) // conditional is true with global isNaN and float is parsed
else query[key] = params[key]
```
Also, we should add comments here about what's going on here because this is a sneaky issue.